src/UI/Components/Diagram/zoomHandler/zoomHandler.ts

```diff
@@ -0,0 +1,307 @@
+import fitToScreen from "../icons/fit-to-screen.svg";
+import requestFullscreen from "../icons/request-fullscreen.svg";
+
+/**
+ * Interface for a button with an icon and a tooltip.
+ *
+ * This interface extends the ui.widgets.button interface and adds methods for setting the icon and the tooltip of the button.
+ */
+interface IconButton extends ui.widgets.button {
+  setIcon(icon: string): void;
+  setTooltip(tooltip: string): void;
+}
+
+/**
+ * IconButton
+ *
+ * It extends the ui.widgets.button class and represents a button with an icon.
+ * It provides methods for setting the icon and the tooltip of the button.
+ *
+ */
+const IconButton = ui.widgets.button.extend({
+  render: function () {
+    const size = this.options.size || 20;
+    const imageEl = document.createElement("img");
+
+    imageEl.style.width = `${size}px`;
+    imageEl.style.height = `${size}px`;
+    this.el.appendChild(imageEl);
+    this.setIcon(this.options.icon);
+    this.setTooltip(this.options.tooltip);
+
+    return this;
+  },
+
+  /**
+   * Sets the icon of the element.
+   *
+   * This method is responsible for setting the source of the image element within the current element to the provided icon.
+   *
+   * @param {string} icon - The source of the icon to set. Defaults to an empty string, which will clear the current icon.
+   */
+  setIcon: function (icon = "") {
+    this.el.querySelector("img").src = icon;
+  },
+
+  /**
+   * Sets the tooltip of the element.
+   *
+   * This method is responsible for setting the tooltip of the current element to the provided tooltip.
+   * It also sets the position of the tooltip.
+   *
+   * @param {string} tooltip - The text of the tooltip to set. Defaults to an empty string, which will clear the current tooltip.
+   * @param {string} direction - The position of the tooltip. Defaults to "right".
+   */
+  setTooltip: function (tooltip = "", direction = "right") {
+    this.el.dataset.tooltip = tooltip;
+    this.el.dataset.tooltipPosition = direction;
+  },
+});
+
+/**
+ * ZoomHandlerService class
+ *
+ * This class is responsible for managing the paning of the canvas.
+ * It provides methods for zoom to fit, zooming in/out in the canvas and fullscreen toggle.
+ *
+ * @class
+ * @prop {ui.Toolbar} toolbar - The toolbar object that contains the paning controls.
+ * @method constructor - Constructs a new instance of the ZoomHandlerService class.
+ * @param {HTMLElement} element - The HTML element that will contain the navigator.
+ * @param {ui.PaperScroller} scroller - The scroller object that allows and zooming in the canvas.
+ */
+export class ZoomHandlerService {
+  toolbar: ui.Toolbar;
+
+  constructor(
+    private element: HTMLElement,
+    private scroller: ui.PaperScroller,
+  ) {
+    this.toolbar = new ui.Toolbar({
+      autoToggle: true,
+      references: {
+        paperScroller: scroller,
+      },
+      tools: [
+        {
+          type: "icon-button",
+          name: "fullscreen",
```

Comment:
It does, is added later, but I'll append it here as well

Reply:
Why would it be added later? isn't it added twice now? 🤔

Reply:
we have the function to update the styling for that button, we can't add the icon in the object, but by setProperty() function, to avoid that as we have logic for that I just run updateStyling function at init.

it won't change anything, that call also will be run and in that way there is no more confusion that some parts don't have tooltip

Reply:
added comments to those places